roxie/udplib/udpipmap.hpp

```diff
@@ -91,15 +91,22 @@ template<class T> class IpMapOf
     }
     T &lookup(const IpAddress &) const;
     inline T &operator[](const IpAddress &ip) const { return lookup(ip); }
-    myIterator begin() { return myIterator(first, firstHash, table); }
+    myIterator begin()
+    {
+        // Take care as it's possible for firstHash to be updated on another thread as we are running
+        unsigned lfirstHash = firstHash;
+        if (lfirstHash==256)
+            return end();
+        else
+            return myIterator(table[lfirstHash].load(std::memory_order_relaxed), lfirstHash, table);
```

Comment:
should be memory order acquire  - otherwise you might end up reading nullptr (not picking up on the store)

Reply:
Changed